### PR TITLE
feat: add list users in group api + groups list cli

### PR DESCRIPTION
**Pull request by pdevine**
## Summary

List users for a given group

## Checklist

- [x] Wrote appropriate unit tests
- [x] Considered security implications of the change
- [x] Updated associated docs where necessary
- [x] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [x] Nothing sensitive logged
- [x] Considered data migrations for smooth upgrades

### CODE REVIEW

File: internal/server/data/group_test.go [hunk 1]
```diff
@@ -126,6 +126,41 @@ func TestListGroups(t *testing.T) {
 			}
 			assert.DeepEqual(t, actual, expected, cmpGroupShallow)
 		})
```
Comment (pdevine) [inline internal/server/data/group_test.go:128]: There are also tests for `internal/server/handlers_test.go` however I realized that there isn't a way to add users to groups yet through the API, so those will have to wait until I add that part.

File: internal/server/data/identity.go [hunk 2]
```diff
@@ -97,6 +97,23 @@ func ListIdentities(db *gorm.DB, selectors ...SelectorFunc) ([]models.Identity,
 	return list[models.Identity](db, selectors...)
 }
 
+func ListIdentitiesByGroup(db *gorm.DB, groupID uid.ID, selectors ...SelectorFunc) ([]models.Identity, error) {
+	group, err := GetGroup(db.Preload("Identities", func(db *gorm.DB) *gorm.DB {
+		for _, selector := range selectors {
```
Comment (pdevine) [inline internal/server/data/identity.go:102]: This adds the `name=?` and `ids=?` selectors to the query, and then sorts the results alphabetically to more closely resemble `ListIdentities()`

File: internal/cmd/groups.go [hunk 3]
```diff
@@ -32,3 +35,75 @@ func createGroup(client *api.Client, name string) (*api.Group, error) {
 
 	return group, nil
 }
+
+func newGroupsCmd(cli *CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Hidden:  true,
```
Comment (pdevine) [inline internal/cmd/groups.go:41]: The command will stay hidden for now until the other subcommands are ready.

File: internal/cmd/groups.go [hunk 4]
```diff
@@ -32,3 +35,75 @@ func createGroup(client *api.Client, name string) (*api.Group, error) {
...
+		Use:     "groups",
+		Short:   "Manage group identities",
```
Comment (BruceMacD) [inline internal/cmd/groups.go:43]:
```suggestion
		Short:   "Manage groups of identities",
```

File: internal/server/data/group_test.go [hunk 5]
```diff
@@ -126,6 +126,41 @@ func TestListGroups(t *testing.T) {
...
+
+		t.Run("filter identities by group", func(t *testing.T) {
```
Comment (BruceMacD) [inline internal/server/data/group_test.go:130]: This should be in `data/identity_test.go`

File: internal/cmd/groups.go (hunk 4 above)
Comment (pdevine) [reply internal/cmd/groups.go:43]: changed.

File: internal/server/data/group_test.go (hunk 5 above)
Comment (pdevine) [reply internal/server/data/group_test.go:130]: moved.

**Review by BruceMacD — APPROVED** [review]

**Pull request merged**